qt/widgets/common/src/AlgorithmHistoryWindow.cpp

```diff
@@ -201,6 +201,16 @@ AlgorithmHistoryWindow::AlgorithmHistoryWindow(
       m_histPropWindow(nullptr), m_execSumGrpBox(nullptr),
       m_envHistGrpBox(nullptr), m_wsName(wsptr->getName().c_str()),
       m_view(wsptr->getHistory().createView()) {
+
```

Comment:
Could I suggest just a simple
```c++
if (m_algHist.empty()) {
   throw std::invalid_argument(
        "No history found on the given workspace");
}
```
rather than needing to move any of the other code?

Reply:
If we are now able to throw in the constructor we will also need to add `throw(std::invalid_argument)` to the end of the constructor declaration in [_common.sip](https://github.com/mantidproject/mantid/blob/master/qt/python/mantidqt/_common.sip#L290). The exception is then properly understood by Python.

Reply:
That makes perfect sense! good call!